frontend/src/lib/components/Dropdown/dropdown.tsx

```diff
@@ -59,74 +59,101 @@ type DropdownRect = {
 
 type OptionItem<TValue> = {
     type: "option";
```

Comment:
Consider using an `enum` as a type here as it is easier to both change/maintain the code and to explicitly find all references.